yeonjae/programmers-2/3차압축.js

```diff
@@ -0,0 +1,58 @@
+function solution(msg) {
+  const dictionary = [
+    "A",
+    "B",
+    "C",
+    "D",
+    "E",
+    "F",
+    "G",
+    "H",
+    "I",
+    "J",
+    "K",
+    "L",
+    "M",
+    "N",
+    "O",
+    "P",
+    "Q",
+    "R",
+    "S",
+    "T",
+    "U",
+    "V",
+    "W",
+    "X",
+    "Y",
+    "Z",
+  ];
+  const result = [];
+
+  while (msg.length !== 0) {
+    let word = "";
+
+    for (let i = 1; i <= msg.length; i++) {
+      const nextWord = msg.substring(0, i);
+
+      if (!dictionary.includes(nextWord)) {
+        dictionary.push(nextWord);
+        break;
+      }
+
+      word = nextWord;
+    }
+
+    result.push(dictionary.lastIndexOf(word) + 1);
+    msg = msg.substr(word.length);
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(msg) {
  const dictionary = Array.from({ length: 26 }, (_, i) =>
    String.fromCharCode(65 + i)
  );
  const result = [];

  while (msg.length !== 0) {
    let word = "";

    for (let i = 1; i <= msg.length; i++) {
      const nextWord = msg.substring(0, i);
      
      if (!dictionary.includes(nextWord)) {
        dictionary.push(nextWord);
        break;
      }
      word = nextWord;
    }
    result.push(dictionary.lastIndexOf(word) + 1);
    msg = msg.substr(word.length);
  }
  return result;
}
```
- 풀이잘봤씁니다
- 하나하나 배열을 정의하기 보다는 Array.from과 fromCharCode로 아스키코드를 문자로 변환해줌으로써 깔끔한 코드가 가능합니다!